pkg/plugins/plugins_darwin.go

```diff
@@ -16,6 +16,20 @@ import (
 	"github.com/newrelic/infrastructure-agent/pkg/sysinfo/hostid"
 )
 
+func heartbeatOnlyMode(a *agent.Agent) bool {
+	config := a.GetContext().Config()
+	if !config.IsSecureForwardOnly && !config.IntegrationsOnly {
```

Comment:
shouldn't be `if IsForwardOnly { ...` to not send the heartbeat on that mode only?

Reply:
Not, the default mode shouldn't send it as well.